models/auth/secret.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package auth
+
+import (
+	"fmt"
+	"regexp"
+
+	"code.gitea.io/gitea/models/db"
+	"code.gitea.io/gitea/modules/timeutil"
+)
+
+type ErrSecretNameInvalid struct {
+	Name string
+}
+
+func (err ErrSecretNameInvalid) Error() string {
+	return fmt.Sprintf("secret name %s is invalid", err.Name)
+}
+
+type ErrSecretDataInvalid struct {
+	Data string
+}
+
+func (err ErrSecretDataInvalid) Error() string {
+	return fmt.Sprintf("secret data %s is invalid", err.Data)
+}
+
+var nameRE = regexp.MustCompile("[^a-zA-Z0-9-_.]+")
+
+// Secret represents a secret
+type Secret struct {
+	ID          int64
+	UserID      int64              `xorm:"index NOTNULL"`
+	RepoID      int64              `xorm:"index NOTNULL"`
+	Name        string             `xorm:"NOTNULL"`
+	Data        string             `xorm:"TEXT"`
+	PullRequest bool               `xorm:"NOTNULL"`
+	CreatedUnix timeutil.TimeStamp `xorm:"created NOTNULL"`
```

Comment:
Should we also store `UpdatedUnix`?
Or should we perhaps **only** store `UpdatedUnix`?